### PR TITLE
The Channels PR

### PR DESCRIPTION
**Pull request by Kqliber**
Added channels.

There's lots of other minor changes to be able to make this work & i cant list them all. most of the commit names should show most of the minor changes, but some might be missing. sorry!

### CODE REVIEW

**Review by Starmism — APPROVED** [review]

**Review by BlitzOffline — APPROVED** [review]

File: plugin/src/main/java/at/helpch/chatchat/util/FormatUtils.java [hunk 1]
```diff
@@ -35,12 +35,13 @@ private FormatUtils() {
...
             @NotNull final Format format,
-            @NotNull final Player player,
+            @NotNull final User user,
             @NotNull final String message) {
         return format.parts().stream()
-                .map(part -> PlaceholderAPI.setPlaceholders(player, part))
+                .map(part -> PlaceholderAPI.setPlaceholders(user.player(), part))
                 .map(part -> part.replace("%message%", message))
-                .map(part -> replaceRecipientPlaceholder(player, part))
+                .map(part -> part.replace("%channel_prefix%", user.channel().channelPrefix()))
```
[the remark this answers is not in the document]
Comment (Kqliber) [reply plugin/src/main/java/at/helpch/chatchat/util/FormatUtils.java:43]: yeah was thinking same, we can do that in the future and add more placeholders ofc, probably worth adding it to the github project board

**Comment by Kqliber** [comment]
users can now see multiple channels at once. This is by using the `chatchat.channel.see.<channel>` permission, and to toggle between the channels, the permission is now `chatchat.channel.use.<channel>`.

when using the message prefix of a channel, it will now respect whether or not you have the correct permission.

At the moment, the <channel> part of the permission currently takes the value from the command name. In the future this will change to use the key (Map.get(key) for example) that's set in the config so its a little bit cleaner.

File: plugin/src/main/java/at/helpch/chatchat/listener/ChatListener.java [hunk 2]
```diff
@@ -31,33 +31,30 @@ public void onChat(final AsyncPlayerChatEvent event) {
...
-        channelByPrefix.ifPresent(channel -> {
-            if (user.canUse(channel)) user.channel(channel);  // set the channel if their message starts with a channel prefix & has perms
-        });
 
         final var message = channelByPrefix.isEmpty()
                 ? event.getMessage()
                 : event.getMessage().replaceFirst(Pattern.quote(channelByPrefix.get().messagePrefix()), "");
 
-        final var newChannel = user.channel();
+        final var channel = user.channel();
```
Comment (BlitzOffline) [inline plugin/src/main/java/at/helpch/chatchat/listener/ChatListener.java:42]: Is this really needed? Maybe instead it should be something like this:
`final var channel = channelByPrefix.isPresent() ? channelByPrefix.get() : user.channel();` or maybe don't put it in a variable at all.

**Review by BlitzOffline — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/at/helpch/chatchat/listener/ChatListener.java (hunk 2 above)
Comment (BlitzOffline) [reply plugin/src/main/java/at/helpch/chatchat/listener/ChatListener.java:42]: Other than this it all seems pretty good. and yes I am nitpick rn lol

**Review by BlitzOffline — APPROVED** [review]

**Pull request merged**